src/sbin/lorax-composer

```diff
@@ -129,6 +131,29 @@ if __name__ == '__main__':
     setup_logging(opts.logfile)
     log.debug("opts=%s", opts)
 
+    # Check to make sure the group exists and get its gid
+    try:
+        gid = grp.getgrnam(opts.group).gr_gid
+    except KeyError:
+        log.error("Missing group '%s'", opts.group)
+        sys.exit(1)
+
+    # Check the socket path to make sure it exists, and that ownership and permissions are correct.
+    socket_dir = os.path.dirname(opts.socket)
+    if not os.path.exists(socket_dir):
+        # Create the directory and set permissions and ownership
+        os.makedirs(socket_dir, 0o750)
+        os.chown(socket_dir, 0, gid)
```

Comment:
I don't think it's necessary to restrict access to the directory. It's always a nuisance not to be able to tab-complete or ls directories when you are typing a sudo command and the like. The mere existence of the socket isn't a secret.

Reply:
I think it's better to err on the side of safety. There's no good reason for anyone else to list the directory.